--- eagle-usb-1.9.6/driver/user/eagle-usb.conf.orig	2004-02-22 22:11:30.000000000 +0100
+++ eagle-usb-1.9.6/driver/user/eagle-usb.conf	2004-06-02 11:30:45.808491752 +0200
@@ -22,7 +22,32 @@
 # OPTN6=00000000
 # OPTN7=02CD8044
 # OPTN15=09090909
-VPI=00000008
+
+# VPI/VCI
+#
+# Country     Network        Encapsulation           VPI VCI (decimal values)
+#
+# Belgium     Belgacom       PPPoA vcmux             8   35
+# France      FT             PPPoA vcmux             8   35
+# France      Free dégroupé  IPoA vcmux              8   36
+# Finland     Sonera         IPoA LLC                0   100
+# Germany     DT             PPPoE LLC               1   32
+# Hungary     Matav          PPPoE LLC               1   32
+# Italy       Telecom Italia PPPoA vcmux             8   35
+# Netherlands KPN            PPPoA vcmux             8   48
+# Poland      TPSA           PPPoA vcmux             0   35
+# Portugal    PT             PPPoE LLC               0   35
+# Spain       Retevision     PPPoA vcmux             8   35
+# Spain       Telefonica     IPoA vcmux ou PPPoE LLC 8   32
+# Sweden      Telia          IPoA LLC                8   35
+# UK          BT             PPPoA vcmux             0   38
+#
+# References:	http://damien.bergamini.free.fr/ueagle/vpivci.html
+#
+# VPI/VCI -	hex. values - if you use values from table above,
+#		you have to convert them from decimal to hex.
+#
+VPI=00000000
 VCI=00000023
 
 #The following values are valid for encapsulation :
