--- eagle-1.0.4/driver/user/adictrl.c.orig	2004-01-18 15:22:15.000000000 +0100
+++ eagle-1.0.4/driver/user/adictrl.c	2004-01-18 15:35:06.439165640 +0100
@@ -181,6 +181,8 @@
  */
 
 typedef struct usb_device_descriptor 
+#ifndef __LINUX_USB_CH9_H
+/* in 2.6 this struct is already available in <linux/usb_ch9.h> */
 {
     uint8_t  bLength;
     uint8_t  bDescriptorType;
@@ -196,7 +198,9 @@
     uint8_t  iProduct;
     uint8_t  iSerialNumber;
     uint8_t  bNumConfigurations;
-} __attribute__((packed)) usb_dev_desc_t;
+} __attribute__((packed))
+#endif
+usb_dev_desc_t;
 
 /**
  * cmv_msg_t - CMV message description.
