--- eagle-usb-2.3.2/driver/user/eagle-usb.conf	2005-03-27 14:45:25.000000000 +0000
+++ eagle-usb-2.3.2.new/driver/user/eagle-usb.conf	2005-06-05 21:40:29.830447256 +0000
@@ -5,7 +5,30 @@
 #
 # VPI / VCI are in hexa 
 # for example, (8, 23) is (8,35) in decimal
-VPI=00000008
+#
+# Country     Network        Encapsulation           VPI VCI (decimal values)
+#
+# Belgium     Belgacom       PPPoA vcmux             8   35
+# France      FT             PPPoA vcmux             8   35
+# France      Free dégroupé  IPoA vcmux              8   36
+# Finland     Sonera         IPoA LLC                0   100
+# Germany     DT             PPPoE LLC               1   32
+# Hungary     Matav          PPPoE LLC               1   32
+# Italy       Telecom Italia PPPoA vcmux             8   35
+# Netherlands KPN            PPPoA vcmux             8   48
+# Poland      TPSA           PPPoA vcmux             0   35
+# Portugal    PT             PPPoE LLC               0   35
+# Spain       Retevision     PPPoA vcmux             8   35
+# Spain       Telefonica     IPoA vcmux ou PPPoE LLC 8   32
+# Sweden      Telia          IPoA LLC                8   35
+# UK          BT             PPPoA vcmux             0   38
+#
+# References:  http://damien.bergamini.free.fr/ueagle/vpivci.html
+#
+# VPI/VCI -    hex. values - if you use values from table above,
+#              you have to convert them from decimal to hex.
+#
+VPI=00000000
 VCI=00000024
 
 #The following values are valid for encapsulation :
