diff -uNr eagle-1.0.4.orig/driver/Makefile eagle-1.0.4/driver/Makefile
--- eagle-1.0.4.orig/driver/Makefile	Wed Jun 18 23:58:41 2003
+++ eagle-1.0.4/driver/Makefile	Wed Jul  9 19:41:39 2003
@@ -106,7 +106,7 @@
 install: driver installfirmware installuser
 	mkdir -p $(INSTALLDIR) && \
 	mkdir -p $(CONFDIR)    && \
-	$(INSTALLER) -m 0664 -o root -g root $(MODULE).o $(INSTALLDIR)/$(MODULE).o && \
+	$(INSTALLER) $(MODULE).o $(DESTDIR)/$(INSTALLDIR)/$(MODULE).o && \
 	/sbin/depmod -a
 
 uninstall:
diff -uNr eagle-1.0.4.orig/driver/firmware/Makefile eagle-1.0.4/driver/firmware/Makefile
--- eagle-1.0.4.orig/driver/firmware/Makefile	Wed Jun  4 01:09:37 2003
+++ eagle-1.0.4/driver/firmware/Makefile	Wed Jul  9 19:42:20 2003
@@ -33,12 +33,12 @@
 	gcc $(CFLAGS) builddsp.c -o buildDSP
 
 install : firmware.bin DSPcode.bin
-	mkdir -p $(CONFIGDIR)
-	$(INSTALLER) -c -m 0664 -o root -g root *.bin $(CONFIGDIR)
+	mkdir -p $(DESTDIR)/$(CONFIGDIR)
+	$(INSTALLER)  *.bin $(DESTDIR)/$(CONFIGDIR)
 
 uninstall:
-	rm -f  $(CONFIGDIR)/DSPcode.bin
-	rm -f $(CONFIGDIR)/firmware.bin
+	rm -f $(DESTDIR)/$(CONFIGDIR)/DSPcode.bin
+	rm -f $(DESTDIR)/$(CONFIGDIR)/firmware.bin
 
 clean :
 	rm -f firmware.bin buildFirmware buildDSP DSPcode.bin
diff -uNr eagle-1.0.4.orig/driver/user/Makefile eagle-1.0.4/driver/user/Makefile
--- eagle-1.0.4.orig/driver/user/Makefile	Wed Jun  4 00:40:37 2003
+++ eagle-1.0.4/driver/user/Makefile	Wed Jul  9 19:40:18 2003
@@ -15,19 +15,19 @@
 	gcc $(CFLAGS) adictrl.c -o adictrl
 
 install : adictrl sagemf800.conf
-	mkdir -p $(CONFIGDIR)
-	$(INSTALLER) -c -m 0755 -o root -g root adictrl $(INSTALLDIR)
-	$(INSTALLER) -c -m 0755 -o root -g root showstat $(INSTALLDIR)
-	$(INSTALLER) -c -m 0755 -o root -g root sagemf800.conf $(CONFIGDIR)/adiusbadsl.conf
-#	$(INSTALLER) -c -m 0755 -o root -g root CMVep.txt $(CONFIGDIR)/
-#	$(INSTALLER) -c -m 0755 -o root -g root CMVei.txt $(CONFIGDIR)/
+	$(INSTALLER) -d $(DESTDIR)/$(CONFIGDIR)
+	$(INSTALLER) adictrl $(DESTDIR)/$(INSTALLDIR)
+	$(INSTALLER) showstat $(DESTDIR)/$(INSTALLDIR)
+	$(INSTALLER) sagemf800.conf $(DESTDIR)/$(CONFIGDIR)/adiusbadsl.conf
+	$(INSTALLER) CMVep.txt $(DESTDIR)/$(CONFIGDIR)
+	$(INSTALLER) CMVei.txt $(DESTDIR)/$(CONFIGDIR)
 
 uninstall:
-	rm -f $(INSTALLDIR)/adictrl
-	rm -f $(INSTALLDIR)/showstat
-	rm -f $(CONFIGDIR)/adiusbadsl.conf
-#	rm -f $(CONFIGDIR)/CMVep.txt
-#	rm -f $(CONFIGDIR)/CMVei.txt
+	rm -f $(DESTDIR)/$(INSTALLDIR)/adictrl
+	rm -f $(DESTDIR)/$(INSTALLDIR)/showstat
+	rm -f $(DESTDIR)/$(CONFIGDIR)/adiusbadsl.conf
+	rm -f $(DESTDIR)/$(CONFIGDIR)/CMVep.txt
+	rm -f $(DESTDIR)/$(CONFIGDIR)/CMVei.txt
 
 
 
diff -uNr eagle-1.0.4.orig/pppoa/Makefile eagle-1.0.4/pppoa/Makefile
--- eagle-1.0.4.orig/pppoa/Makefile	Thu Apr 24 20:42:06 2003
+++ eagle-1.0.4/pppoa/Makefile	Wed Jul  9 19:35:04 2003
@@ -45,10 +45,10 @@
 	gcc $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o debug.o debug.c
 
 install: pppoa
-	$(INSTALLER) -m 775 -o root -g root pppoa /usr/sbin
+	$(INSTALLER) pppoa $(DESTDIR)/usr/sbin
 
 uninstall:
-	rm -f /usr/sbin/pppoa
+	rm -f $(DESTDIR)/usr/sbin/pppoa
 
 clean:
 	rm -f *.o pppoa 
diff -uNr eagle-1.0.4.orig/scripts/Makefile eagle-1.0.4/scripts/Makefile
--- eagle-1.0.4.orig/scripts/Makefile	Wed Jun 18 23:58:02 2003
+++ eagle-1.0.4/scripts/Makefile	Wed Jul  9 19:44:40 2003
@@ -14,29 +14,29 @@
 	./makescripts
 
 install: userscripts
-	mkdir -p $(HOTPLUGDIR)/usb
+	mkdir -p $(DESTDIR)/$(HOTPLUGDIR)/usb
 	@if [ -f pap-secrets ]; then \
-		$(INSTALLER) -c -m 0600 -o root -g root pap-secrets $(PPPDIR) ; \
+		$(INSTALLER) -c pap-secrets $(DESTDIR)/$(PPPDIR) ; \
 	fi
 	@if [ -f chap-secrets ]; then \
-		$(INSTALLER) -c -m 0600 -o root -g root chap-secrets $(PPPDIR) ; \
+		$(INSTALLER) -c chap-secrets $(DESTDIR)/$(PPPDIR) ; \
 	fi
 	@if [ -f options.adsl ]; then \
-		$(INSTALLER) -c -m 0644 -o root -g root options.adsl $(PPPDIR) ; \
+		$(INSTALLER) -c options.adsl $(DESTDIR)/$(PPPDIR) ; \
 	fi
-	$(INSTALLER) -c -m 0554 -o root -g root hotplug/adiusb* $(HOTPLUGDIR)/usb
+	$(INSTALLER) -c hotplug/adiusb* $(DESTDIR)/$(HOTPLUGDIR)/usb
 	@if [ -f usb.usermap ]; then \
-		$(INSTALLER) -c -m 0644 -o root -g root usb.usermap $(HOTPLUGDIR) ; \
+		$(INSTALLER) -c usb.usermap $(DESTDIR)/$(HOTPLUGDIR) ; \
 	else \
-		$(INSTALLER) -c -m 0644 -o root -g root usb.usermap.local $(HOTPLUGDIR) && \
+		$(INSTALLER) -c usb.usermap.local $(DESTDIR)/$(HOTPLUGDIR) && \
 		/usr/sbin/update-usb.usermap ; \
 	fi
-	$(INSTALLER) -c -m 0554 -o root -g root adsl/* $(INSTALLDIR)
+	$(INSTALLER) -c  adsl/* $(DESTDIR)/$(INSTALLDIR)
 	rm -f options.adsl options.mire pap-secrets chap-secrets usb.usermap usb.usermap.local
 
 uninstall:
-	rm -f $(PPPDIR)/options.mire
-	rm -f $(HOTPLUGDIR)/usb/adiusb*
+	rm -f $(DESTDIR)/$(PPPDIR)/options.mire
+	rm -f $(DESTDIR)/$(HOTPLUGDIR)/usb/adiusb*
 	./scriptUninst
 
 clean:
