--- driver/Boot.c~	2004-02-22 21:11:30.000000000 +0000
+++ driver/Boot.c	2004-06-02 08:57:53.063519192 +0000
@@ -57,6 +57,15 @@
 static void IDMACompletion(struct urb *urb, struct pt_regs *regs );
 #endif
 
+static inline void wait_ms(unsigned int ms) {
+	        if (!in_interrupt()) {
+	                set_current_state(TASK_UNINTERRUPTIBLE);
+	                schedule_timeout(1 + ms * HZ / 1000);
+	        }
+	        else
+	                mdelay(ms);
+}
+
 /*********************************************************************************/
 /* BootModem									 */
 /* Kinda like doing a ctrl-alt-delete on your PC. Reuploads (or does first	 */

