--- eagle-usb/pppoa/Makefile.orig	2003-11-17 18:08:16.000000000 +0100
+++ eagle-usb/pppoa/Makefile	2003-11-17 18:08:28.000000000 +0100
@@ -45,7 +45,7 @@
 	gcc $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o debug.o debug.c
 
 install: pppoa
-	$(INSTALL) -m 775 -o root -g root pppoa $(SBINDIR)
+	$(INSTALL) pppoa $(SBINDIR)
 
 uninstall:
 	rm -f /usr/sbin/pppoa
--- eagle-usb/driver/user/Makefile.orig	2003-11-17 18:05:56.000000000 +0100
+++ eagle-usb/driver/user/Makefile	2003-11-17 18:06:23.000000000 +0100
@@ -17,9 +17,9 @@
 install : eaglectrl eagle-usb.conf
 	mkdir -p $(INSTALLDIR)
 	mkdir -p $(EAGLEUSB_CONFDIR)
-	$(INSTALL) -m 0755 -o root -g root eaglectrl $(INSTALLDIR)
-	$(INSTALL) -m 0755 -o root -g root eaglestat $(INSTALLDIR)
-	$(INSTALL) -m 0644 -o root -g root eagle-usb.conf $(EAGLEUSB_CONFDIR)/eagle-usb.conf
+	$(INSTALL) eaglectrl $(INSTALLDIR)
+	$(INSTALL) eaglestat $(INSTALLDIR)
+	$(INSTALL) eagle-usb.conf $(EAGLEUSB_CONFDIR)/eagle-usb.conf
 #	$(INSTALL) -m 0755 -o root -g root CMVep.txt $(CONFIGDIR)/
 #	$(INSTALL) -m 0755 -o root -g root CMVei.txt $(CONFIGDIR)/
 
--- eagle-usb/driver/firmware/Makefile.orig	2003-11-17 18:03:24.000000000 +0100
+++ eagle-usb/driver/firmware/Makefile	2003-11-17 18:03:37.000000000 +0100
@@ -36,7 +36,7 @@
 
 install : dsp_code_pots.bin dsp_code_isdn.bin 
 	mkdir -p $(EAGLEUSB_BINDIR)
-	$(INSTALL) -m 0664 -o root -g root $^ $(EAGLEUSB_BINDIR)
+	$(INSTALL) $^ $(EAGLEUSB_BINDIR)
 
 uninstall:
 	rm -f $(EAGLEUSB_BINDIR)/dsp_code_pots.bin $(EAGLEUSB_BINDIR)/dsp_code_isdn.bin
