--- eagle-usb/driver/user/eaglectrl.c.orig	2004-01-13 23:49:59.646045512 +0100
+++ eagle-usb/driver/user/eaglectrl.c	2004-01-13 23:50:45.165125560 +0100
@@ -54,13 +54,8 @@
 
 #define NB_SUPPORTED_PRODS  (sizeof(supported_products)/sizeof(supported_products[0]))
 
-#ifndef CONF_DIR
 #define CONF_DIR            "/etc/eagle-usb"
-#endif
-
-#ifndef BIN_DIR
-#define BIN_DIR             "/etc/eagle-usb"
-#endif
+#define BIN_DIR             "/usr/share/misc"
 
 #define ISDN_BIN_FILE    "dsp_code_isdn.bin"
 #define POTS_BIN_FILE    "dsp_code_pots.bin"
