diff -urN eagle-usb-2.0.0.orig/driver/eu_eth.c eagle-usb-2.0.0.patch/driver/eu_eth.c
--- eagle-usb-2.0.0.orig/driver/eu_eth.c	2004-10-29 19:29:39.000000000 +0200
+++ eagle-usb-2.0.0.patch/driver/eu_eth.c	2005-01-21 09:05:55.605326640 +0100
@@ -34,7 +34,9 @@
 #include "eu_eth.h"
 #include "eu_sm.h"
 
-
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+#define usb_unlink_urb usb_kill_urb
+#endif
 
 /* --------------------- Private Functions Declaration --------------------- */
 static int eu_eth_open ( struct net_device *dev );
diff -urN eagle-usb-2.0.0.orig/driver/eu_main.c eagle-usb-2.0.0.patch/driver/eu_main.c
--- eagle-usb-2.0.0.orig/driver/eu_main.c	2004-11-07 10:06:55.000000000 +0100
+++ eagle-usb-2.0.0.patch/driver/eu_main.c	2005-01-21 09:06:56.187116808 +0100
@@ -53,6 +53,10 @@
 #include "eu_eth.h"
 #include "eu_boot_sm.h"
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+#define usb_unlink_urb usb_kill_urb
+#endif
+
 /* ------------------------------- Private Macros ------------------------------- */
 
 #define CASE_PREFIRM	EAGLE_I_PID_PREFIRM:    \
diff -urN eagle-usb-2.0.0.orig/driver/eu_utils.c eagle-usb-2.0.0.patch/driver/eu_utils.c
--- eagle-usb-2.0.0.orig/driver/eu_utils.c	2004-11-07 10:06:55.000000000 +0100
+++ eagle-usb-2.0.0.patch/driver/eu_utils.c	2005-01-21 09:07:36.400003520 +0100
@@ -31,6 +31,9 @@
 #include "debug.h"
 #include "eu_firmware.h"
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+#define usb_unlink_urb usb_kill_urb
+#endif
 
 /* ----------------------- Private Macros/Variables ------------------------ */
 
diff -urN eagle-usb-2.0.0.orig/driver/Pipes.c eagle-usb-2.0.0.patch/driver/Pipes.c
--- eagle-usb-2.0.0.orig/driver/Pipes.c	2004-10-17 22:35:55.000000000 +0200
+++ eagle-usb-2.0.0.patch/driver/Pipes.c	2005-01-21 09:07:57.586782640 +0100
@@ -35,6 +35,10 @@
 #include "debug.h"
 #include "eu_sm.h"
 
+#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,9)
+#define usb_unlink_urb usb_kill_urb
+#endif
+
 /**
  * --| Reception Mechanism:
  *
