diff -uNr eagle-usb-2.3.2.orig/driver/user/eagle-usb.conf eagle-usb-2.3.2/driver/user/eagle-usb.conf
--- eagle-usb-2.3.2.orig/driver/user/eagle-usb.conf	2005-03-27 16:45:25.000000000 +0200
+++ eagle-usb-2.3.2/driver/user/eagle-usb.conf	2006-01-19 20:55:04.766007000 +0100
@@ -5,8 +5,32 @@
 #
 # VPI / VCI are in hexa 
 # for example, (8, 23) is (8,35) in decimal
-VPI=00000008
-VCI=00000024
+#
+# Country     Network        Encapsulation           VPI VCI (decimal values)
+#
+# Belgium     Belgacom       PPPoA vcmux             8   35
+# France      FT             PPPoA vcmux             8   35
+# France      Free dégroupé  IPoA vcmux              8   36
+# Finland     Sonera         IPoA LLC                0   100
+# Germany     DT             PPPoE LLC               1   32
+# Hungary     Matav          PPPoE LLC               1   32
+# Italy       Telecom Italia PPPoA vcmux             8   35
+# Netherlands KPN            PPPoA vcmux             8   48
+# Poland      TPSA           PPPoA vcmux             0   35
+# Portugal    PT             PPPoE LLC               0   35
+# Spain       Retevision     PPPoA vcmux             8   35
+# Spain       Telefonica     IPoA vcmux ou PPPoE LLC 8   32
+# Sweden      Telia          IPoA LLC                8   35
+# UK          BT             PPPoA vcmux             0   38
+#
+# References:  http://damien.bergamini.free.fr/ueagle/vpivci.html
+#
+# VPI/VCI -    hex. values - if you use values from table above,
+#              you have to convert them from decimal to hex.
+#
+# this is for tpnet neostrada
+VPI=00000000
+VCI=00000023
 
 #The following values are valid for encapsulation :
 #MPOA_MODE_BRIDGED_ETH_LLC ----> 1 RFC2516 Bridged PPPoE LLC
@@ -15,9 +39,16 @@
 #MPOA_MODE_ROUTED_IP_VC    ----> 4 RFC1483/2684 Routed IP (IPoA) VCmux
 #MPOA_MODE_PPPOA_LLC       ----> 5 RFC2364 PPPoA LLC
 #MPOA_MODE_PPPOA_VC        ----> 6 RFC2364 PPPoA VCmux
-Encapsulation=00000004
+#Encapsulation=00000004
+# for TPNET neostrada
+Encapsulation=00000006
 
 # A for CMV use, 1 otherwise
 Linetype=0000000A
 RatePollFreq=00000009
 </eaglectrl>
+# for TPNET neostrada
+STATIC_IP=none
+ISP=PL01
+LANG=
+ASYNCHRONOUS_START=1
