diff -urN eagle-usb-2.0.0.org/driver/eu_types.h eagle-usb-2.0.0/driver/eu_types.h
--- eagle-usb-2.0.0.org/driver/eu_types.h	2004-11-07 10:06:55.000000000 +0100
+++ eagle-usb-2.0.0/driver/eu_types.h	2004-11-25 22:58:46.196401236 +0100
@@ -156,13 +156,13 @@
  */
 typedef struct 
 {
-    uint8_t bRequestType;
-    uint8_t bRequest;
-    uint16_t wValue;
-    uint16_t wIndex;
-    uint16_t wLength;
-    uint8_t data[20];
-} __attribute__ ((packed)) eu_cdc_t;
+    uint8_t bRequestType __attribute__ ((packed));
+    uint8_t bRequest __attribute__ ((packed));
+    uint16_t wValue __attribute__ ((packed));
+    uint16_t wIndex __attribute__ ((packed));
+    uint16_t wLength __attribute__ ((packed));
+    uint8_t data[20] __attribute__ ((packed));
+} eu_cdc_t;
 
 
 /**
