--- eagle-usb/driver/user/eaglectrl.c.orig	2004-01-13 23:49:59.646045512 +0100
+++ eagle-usb/driver/user/eaglectrl.c	2004-01-13 23:50:45.165125560 +0100
@@ -54,13 +54,8 @@
 
 #define NB_SUPPORTED_PRODS  (sizeof(supported_products)/sizeof(supported_products[0]))
 
-#ifndef CONF_DIR
 #define CONF_DIR            "/etc/eagle-usb"
-#endif
-
-#ifndef BIN_DIR
-#define BIN_DIR             "/etc/eagle-usb"
-#endif
+#define BIN_DIR             "/usr/share/misc"
 
 #define ISDN_BIN_FILE    "dsp_code_isdn.bin"
 #define POTS_BIN_FILE    "dsp_code_pots.bin"
--- eagle-usb.org/configure.in	2004-01-13 23:10:20.000000000 +0100
+++ eagle-usb/configure.in	2004-02-04 15:35:29.078214816 +0100
@@ -207,10 +207,10 @@
 
 # ===== display warning(s) =====
 echo
-if test "${KERNELVER}" = "none" ; then
-	echo -e ${KERNEL_SRC_MISSING_MSG}
-	exit 1
-fi
+#if test "${KERNELVER}" = "none" ; then
+#	echo -e ${KERNEL_SRC_MISSING_MSG}
+#	exit 1
+#fi
 CURKERNELVER="`uname -r`"
 if test "${KERNELVER}" != "${CURKERNELVER}" ; then
 	echo -e ${KERNEL_SRC_DIFF_MSG}
