--- eagle-usb/pppoa/Makefile.orig	2003-11-17 18:08:16.000000000 +0100
+++ eagle-usb/pppoa/Makefile	2003-11-17 18:08:28.000000000 +0100
@@ -45,7 +45,7 @@
 	gcc $(CFLAGS) '-DVERSION="$(VERSION)"' -c -o debug.o debug.c
 
 install: pppoa
-	$(INSTALL) -m 775 -o root -g root pppoa $(SBINDIR)
+	$(INSTALL) pppoa $(SBINDIR)
 
 uninstall:
 	rm -f /usr/sbin/pppoa
--- eagle-usb/driver/user/Makefile.orig	2003-11-17 18:05:56.000000000 +0100
+++ eagle-usb/driver/user/Makefile	2003-11-17 18:06:23.000000000 +0100
@@ -17,9 +17,9 @@
 install : eaglectrl eagle-usb.conf
 	mkdir -p $(INSTALLDIR)
 	mkdir -p $(EAGLEUSB_CONFDIR)
-	$(INSTALL) -m 0755 -o root -g root eaglectrl $(INSTALLDIR)
-	$(INSTALL) -m 0755 -o root -g root eaglestat $(INSTALLDIR)
-	$(INSTALL) -m 0644 -o root -g root eagle-usb.conf $(EAGLEUSB_CONFDIR)/eagle-usb.conf
+	$(INSTALL) eaglectrl $(INSTALLDIR)
+	$(INSTALL) eaglestat $(INSTALLDIR)
+	$(INSTALL) eagle-usb.conf $(EAGLEUSB_CONFDIR)/eagle-usb.conf
 #	$(INSTALL) -m 0755 -o root -g root CMVep.txt $(CONFIGDIR)/
 #	$(INSTALL) -m 0755 -o root -g root CMVei.txt $(CONFIGDIR)/
 
--- eagle-usb/driver/firmware/Makefile.orig	2003-11-17 18:03:24.000000000 +0100
+++ eagle-usb/driver/firmware/Makefile	2003-11-17 18:03:37.000000000 +0100
@@ -36,7 +36,7 @@
 
 install : dsp_code_pots.bin dsp_code_isdn.bin 
 	mkdir -p $(EAGLEUSB_BINDIR)
-	$(INSTALL) -m 0664 -o root -g root $^ $(EAGLEUSB_BINDIR)
+	$(INSTALL) $^ $(EAGLEUSB_BINDIR)
 
 uninstall:
 	rm -f $(EAGLEUSB_BINDIR)/dsp_code_pots.bin $(EAGLEUSB_BINDIR)/dsp_code_isdn.bin
--- eagle-usb/configure.in~	2003-12-13 17:48:16.000000000 +0100
+++ eagle-usb/configure.in	2003-12-13 18:02:10.147697088 +0100
@@ -40,7 +40,7 @@
 AC_ARG_WITH(lang,
 [  --with-lang=LL             Language (en/de/es/fr/it/pl)],
 EU_LANG=${withval},
-EU_LANG=${LANGUAGE:0:2}
+EU_LANG=${LANGUAGE}
 )
 if test -z "${EU_LANG}" ; then
 	EU_LANG="en"
--- eagle-usb/utils/scripts/Makefile.orig	2003-12-15 16:33:08.582485448 +0100
+++ eagle-usb/utils/scripts/Makefile	2003-12-15 16:35:59.179550752 +0100
@@ -59,7 +59,6 @@
 	$(INSTALL) -m 0755 -o root -g root tmp/eu_init $(EU_SCRIPT_DIR) && \
 	$(INSTALL) -m 0644 -o root -g root usb.usermap $(EU_SCRIPT_DIR) && \
 	$(INSTALL) -m 0644 -o root -g root options $(EU_SCRIPT_DIR) && \
-	cp -R lang/* $(EU_LANG_DIR) && \
 	rm -Rf $(EU_LANG_DIR)/CVS && \
 	touch $(EU_SCRIPT_DIR)/lock
 #	eagle-usb.conf => installé par ~/driver/user/Makefile
--- eagle-usb/configure.in.orig	2003-12-19 15:11:37.000000000 +0100
+++ eagle-usb/configure.in	2003-12-19 15:15:39.000000000 +0100
@@ -36,6 +36,14 @@
 EAGLEUSB_CONFDIR=${CONFDIR}/eagle-usb
 )
 
+# ===== arg: eagle-usb-confdir =====
+AC_ARG_WITH(eagle-usb-bindir,
+[  --with-eagle-usb-bindir   Path to EAGLE USB configuration files ($DATADIR/eagle-usb) ],
+EAGLEUSB_BINDIR=${withval},
+EAGLEUSB_BINDIR=${DATADIR}/eagle-usb
+)
+
+
 # ===== arg: language =====
 AC_ARG_WITH(lang,
 [  --with-lang=LL             Language (en/de/es/fr/it/pl)],
--- eagle-usb/configure.in~	2003-12-19 23:31:13.000000000 +0100
+++ eagle-usb/configure.in	2003-12-19 23:58:46.000000000 +0100
@@ -181,6 +181,7 @@
 AC_SUBST(KERNELVER)
 AC_SUBST(CONFDIR)
 AC_SUBST(EAGLEUSB_CONFDIR)
+AC_SUBST(EAGLEUSB_BINDIR)
 AC_SUBST(EU_LANG)
 AC_SUBST(DIALOG)
 AC_SUBST(DHCP)
