--- eagle-usb-1.9.6/driver/Sm.c~	2004-03-10 00:31:35.000000000 +0100
+++ eagle-usb-1.9.6/driver/Sm.c	2004-03-10 00:31:35.000000000 +0100
@@ -877,8 +877,8 @@
                         ENDCASE;
                     case 9:  /* RATE.actual*/
                         pAdiSM->XferRate0 = pMsg->data;
-                        pAdiSM->DownRate = le16_to_cpu (pMsg->data>>16) *32*1024;
-                        pAdiSM->UpRate = le16_to_cpu(pMsg->data&0xFFFF) *32*1024;
+                        pAdiSM->DownRate = (pMsg->data>>16) *32*1024;
+                        pAdiSM->UpRate = (pMsg->data&0xFFFF) *32*1024;
                         
                         if ( !EU_TEST_FLAG(ins,EU_LOW_RATE) )
                         {
