--- eagle-usb-1.9.6/driver/eu_main.c~	2004-05-10 08:42:57.249887328 +0200
+++ eagle-usb-1.9.6/driver/eu_main.c	2004-05-10 08:42:57.249887328 +0200
@@ -476,11 +476,9 @@
     eu_enters (DBG_INIT);
 
 #ifdef LINUX_2_6
-        if ( intf->driver != NULL )
-        {
-            ins = usb_get_intfdata ( intf );
+        ins = usb_get_intfdata ( intf );
+        if ( ins != NULL )
             goto byebye;
-        }
 #endif
         
     /*
