--- eagle-1.0.4/driver/user/adictrl.c~	2003-06-03 22:40:57.000000000 +0000
+++ eagle-1.0.4/driver/user/adictrl.c	2003-07-15 19:49:44.000000000 +0000
@@ -54,8 +54,8 @@
 
 #define NB_SUPPORTED_PRODS  (sizeof(supported_products)/sizeof(supported_products[0]))
 
-#define DEFAULT_FIRMWAREFILE "/etc/analog/firmware.bin"
-#define DEFAULT_DSPFILE	     "/etc/analog/DSPcode.bin"
+#define DEFAULT_FIRMWAREFILE "/usr/share/misc/firmware.bin"
+#define DEFAULT_DSPFILE	     "/usr/share/misc/DSPcode.bin"
 #define DEFAULT_OPTIONSFILE  "/etc/analog/adiusbadsl.conf"
 
 #if 0
--- eagle-1.0.4/driver/firmware/Makefile~	2003-07-15 19:43:48.000000000 +0000
+++ eagle-1.0.4/driver/firmware/Makefile	2003-07-15 19:47:24.000000000 +0000
@@ -33,8 +33,8 @@
 	gcc $(CFLAGS) builddsp.c -o buildDSP
 
 install : firmware.bin DSPcode.bin
-	mkdir -p $(DESTDIR)/$(CONFIGDIR)
-	$(INSTALLER)  *.bin $(DESTDIR)/$(CONFIGDIR)
+	mkdir -p $(DESTDIR)/usr/share/misc
+	$(INSTALLER)  *.bin $(DESTDIR)/usr/share/misc
 
 uninstall:
 	rm -f $(DESTDIR)/$(CONFIGDIR)/DSPcode.bin
diff -urN aaa/eagle-1.0.4/readme.txt eagle-1.0.4/readme.txt
--- aaa/eagle-1.0.4/readme.txt	2003-06-18 23:58:25.000000000 +0200
+++ eagle-1.0.4/readme.txt	2003-07-15 22:22:10.000000000 +0200
@@ -146,7 +146,7 @@
     /usr/sbin/stopadsl
     /usr/sbin/showstat
     /etc/analog/adiusbadsl.conf
-    /etc/analog/DSPcode.bin
+    /usr/share/misc/DSPcode.bin
     /etc/hotplug/usb/adiusbfirmware
     /etc/hotplug/usb/adiusbdsp
     /etc/ppp/options.adsl
@@ -173,7 +173,7 @@
 
 where "file" is the binary file containing the DSP code. This file can
 be generated with the tool driver/firmware/buildDSP from the .BNM files.
-The default DSP code file is the file /etc/analog/DSPcode.bin.
+The default DSP code file is the file /usr/share/misc/DSPcode.bin.
     The module keeps in memory the DSP code, since the modem can need
 it at any time. However, it is not resent automatically when the modem
 is plugged in, so the firmware must be reloaded explicitely in this case.
